src/cli/service.rs

```diff
@@ -247,7 +235,6 @@ impl Service {
         shutdown_rx: &crate::signal::ShutdownRx,
     ) -> crate::Result<tokio::task::JoinHandle<crate::Result<()>>> {
         let mut shutdown_rx = shutdown_rx.clone();
-        let mds_task = self.publish_mds(config)?;
```

Comment:
We still need to keep the old path in until this change has been rolled out.

I think it would make sense to spawn a separate task that does both when you provide `--service.grpc`, then once we've rolled out and switched the infra to use that task, we can remove all code in the `--service.xDS` and `--service.mds` paths.